site/src/pages/[shortname]/blocks/[blockSlug].page.tsx

```diff
@@ -123,25 +136,32 @@ export const getStaticProps: GetStaticProps<
     return { notFound: true };
   }
 
+  // getStaticProps doesn't parse undefined, so returning an empty string instead
+  const repositoryDisplayUrl = blockMetadata.repository
```

Comment:
Decided to generate the display URL in `getStaticProps` since we can generate it in one-shot here, without ensuring any props-based reactivity.

Reply:
My preference would be to call this transform as close as possible to where we display the link. However, I’m not strongly pro this. Ideally, initial page props should be orthogonal to each other, i.e. completely independent. Anything we can derive from them can go into components and we can use `React.memo` / `useMemo` if needed for some heavy stuff. That’s not the case here.

Because `repositoryDisplayUrl` can be derived from `blockMetadata`, our page data gets bigger but does not carry more information. Besides, if we decide to change the page logic later, it is more likely that we’ll keep the above lines but not use the result anywhere.

Reply:
> Ideally, initial page props should be orthogonal to each other, i.e. completely independent.

Makes sense.